package/v4l2tools/02-enable-cross-compile.patch

```diff
@@ -1,118 +1,153 @@
- v4l2copy: src/v4l2copy.cpp  libv4l2wrapper.a
-@@ -126,13 +117,13 @@ h264bitstream/Makefile:
- 	cd h264bitstream && autoreconf -i -f && ./configure --host $(shell $(CC) -dumpmachine)
- 
- h264bitstream/.libs/libh264bitstream.so: h264bitstream/Makefile
--	make -C h264bitstream 
-+	make CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" -C h264bitstream
- 
- hevcbitstream/Makefile:
- 	cd hevcbitstream && autoreconf -i -f && LDFLAGS=-lm ./configure --host $(shell $(CC) -dumpmachine)
- 
- hevcbitstream/.libs/libhevcbitstream.so: hevcbitstream/Makefile
--	make -C hevcbitstream 
-+	make CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" -C hevcbitstream
- 
- v4l2dump: src/v4l2dump.cpp libv4l2wrapper.a h264bitstream/.libs/libh264bitstream.so  hevcbitstream/.libs/libhevcbitstream.so libyuv.a
- 	$(CXX) -o $@ $(CFLAGS) $^ $(LDFLAGS) -Ih264bitstream  -Ihevcbitstream -Wl,-rpath=./h264bitstream/.libs,-rpath=./hevcbitstream/.libs -I libyuv/include 
++ALL_PROGS+=v4l2fuse
++endif
++
++all: $(ALL_PROGS)
++
++# read V4L2 capture -> write V4L2 output
++v4l2copy: src/v4l2copy.cpp
++	$(CXX) -o $@ $(CXXFLAGS) $^ $(LDFLAGS)
++
++# read V4L2 capture -> convert YUV format -> write V4L2 output
++v4l2convert_yuv: src/v4l2convert_yuv.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lyuv -ljpeg
++
++# -> write V4L2 output
++v4l2source_yuv: src/v4l2source_yuv.cpp
++	$(CXX) -o $@ $(CXXFLAGS) $^ $(LDFLAGS)
++
++# read V4L2 capture -> compress using libvpx -> write V4L2 output
++v4l2compress_vpx: src/v4l2compress_vpx.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lvpx -lyuv -ljpeg
++
++# read V4L2 capture -> compress using x264 -> write V4L2 output
++v4l2compress_h264: src/v4l2compress_h264.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lh264bitstream -lyuv -ljpeg
++
++# read V4L2 capture -> compress using x265 -> write V4L2 output
++v4l2compress_x265: src/v4l2compress_x265.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lhevcbitstream -lyuv -ljpeg
++
++# read V4L2 capture -> compress using libjpeg -> write V4L2 output
++v4l2compress_jpeg: src/v4l2compress_jpeg.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lyuv -ljpeg
++
++# read V4L2 capture -> uncompress using libjpeg -> write V4L2 output
++v4l2uncompress_jpeg: src/v4l2uncompress_jpeg.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lyuv -ljpeg
++
++# try with opencv
++v4l2detect_yuv: src/v4l2detect_yuv.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lopencv_core -lopencv_objdetect -lopencv_imgproc -lyuv -ljpeg
++
++# dump
++v4l2dump: src/v4l2dump.cpp
++	$(CXX) -o $@ $(CXXFLAGS) -I$(PREFIX)/include/h264bitstream -I$(PREFIX)/include/hevcbitstream -I$(PREFIX)/include/libyuv $^ $(LDFLAGS) -lh264bitstream -lhevcbitstream -lyuv -ljpeg
++
++v4l2fuse: src/v4l2fuse.c
++	$(CC) -o $@ $(CXXFLAGS) $^ $(LDFLAGS) -D_FILE_OFFSET_BITS=64 -lfuse
```

Comment:
Compilation fails at `v4l2fuse.c` with `error: array type has incomplete element type`. To fix, use C compiler for v4l2fuse:
```
$(CC) -o $@ $(CFLAGS) $^ $(LDFLAGS) -D_FILE_OFFSET_BITS=64 -lfuse
```